### PR TITLE
Call GetNewUnit for OPEN(NEWUNIT=scalar-int-variable) statements

### PR DESCRIPTION
**Pull request by vdonaldson**
An `OPEN(NEWUNIT=unit)` statement generates a runtime unit number.
Call `GetNewUnit` to assign the number to `unit`.

### CODE REVIEW

**Review by psteinfeld — APPROVED** [review]
All builds, tests, and looks good.

File: flang/lib/Lower/IO.cpp [hunk 1]
```diff
@@ -941,6 +941,25 @@ mlir::Value genIOOption<Fortran::parser::ConnectSpec::Recl>(
...
+    mlir::Value cookie, const Fortran::parser::ConnectSpec::Newunit &spec) {
+  Fortran::lower::StatementContext stmtCtx;
+  fir::FirOpBuilder &builder = converter.getFirOpBuilder();
+  mlir::FuncOp ioFunc = getIORuntimeFunc<mkIOKey(GetNewUnit)>(loc, builder);
+  mlir::FunctionType ioFuncTy = ioFunc.getType();
+  const auto *var = Fortran::semantics::GetExpr(spec);
+  mlir::Value addr = builder.createConvert(
+      loc, ioFuncTy.getInput(1),
+      fir::getBase(converter.genExprAddr(var, stmtCtx, loc)));
+  auto kind = builder.create<mlir::arith::ConstantOp>(
```
Comment (jeanPerier) [inline flang/lib/Lower/IO.cpp:956]: Using `builder.createConstantInteger` here would hide the attribute handling part.

**Review by jeanPerier — COMMENTED** [review]
Would it make sense to remove the default template case in `genIOOption` (and to make it a non template function) to ensure we do not miss any cases / are aware that something needs to be done if any new ones are added ?

**Review by schweitzpgi — COMMENTED** [review]
LGTM.

File: flang/lib/Lower/IO.cpp (hunk 1 above)
Comment (vdonaldson) [reply flang/lib/Lower/IO.cpp]: That's better code; thanks

**Review by jeanPerier — APPROVED** [review]
Looks great, thanks Val

**Pull request merged**